genie-web/src/main/java/com/netflix/genie/web/services/impl/JobOwnerServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *  Copyright 2016 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.services.impl;
+
+import com.netflix.genie.common.exceptions.GenieException;
+import com.netflix.genie.common.exceptions.GenieNotFoundException;
+import com.netflix.genie.web.services.AgentRoutingService;
+import com.netflix.genie.web.services.JobOwnerService;
+import com.netflix.genie.web.services.JobPersistenceService;
+import com.netflix.genie.web.services.JobSearchService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import javax.validation.constraints.NotBlank;
+
+/**
+ * Service to get information about the server owning a job.
+ * Job can be a v3 job which runs locally on a genie node. This node is the job owner.
+ * Job can be a v4 job running remotely on a agent. The genie node which has the
+ * connection to this agent is the owner.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+@Slf4j
+@Service
```

Comment:
This should be defined in `GenieServicesAutoConfiguration` with a `ConditionalOnMissingBean` like the rest of our services so it can be easily overridden.